SpatialGDK/Source/SpatialGDKEditor/Private/SpatialTestSettings.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialTestSettings);
+
+FSpatialTestSettings::FSpatialTestSettings()
+	: OriginalSpatialGDKSettings(nullptr)
+	, OriginalLevelEditorPlaySettings(nullptr)
+	, OriginalSpatialGDKEditorSettings(nullptr)
+	, OriginalGeneralProjectSettings(nullptr)
+	, OriginalEditorPerformanceSettings(nullptr)
+{
+}
+
+void FSpatialTestSettings::Override(const FString& MapName)
+{
+	// Back up the existing settings so they can be reverted later
+	Duplicate<USpatialGDKSettings>(OriginalSpatialGDKSettings);
+	Duplicate<ULevelEditorPlaySettings>(OriginalLevelEditorPlaySettings);
+	Duplicate<USpatialGDKEditorSettings>(OriginalSpatialGDKEditorSettings);
+	Duplicate<UGeneralProjectSettings>(OriginalGeneralProjectSettings);
+	Duplicate<UEditorPerformanceSettings>(OriginalEditorPerformanceSettings);
+	// First override the settings from the base config file, if it exists
+	if (FPaths::FileExists(BaseOverridesFilename))
+	{
+		Load(BaseOverridesFilename);
+	}
+	else
+	{
+		UE_LOG(LogSpatialTestSettings, Log, TEXT("Base config file not found %s."), *BaseOverridesFilename);
```

Comment:
Should this be an error?

Reply:
This is not an error, as all the config files for test settings overrides are optional. We needed logging for both the positive and negative cases to find out why the CI was not passing.